src/main/java/org/opensearch/securityanalytics/model/IocDao.java

```diff
@@ -0,0 +1,274 @@
+                case CREATED_FIELD:
+                    if (xcp.currentToken() == XContentParser.Token.VALUE_NULL) {
+                        created = null;
+                    } else if (xcp.currentToken().isValue()) {
+                        created = Instant.ofEpochMilli(xcp.longValue());
+                    } else {
+                        XContentParserUtils.throwUnknownToken(xcp.currentToken(), xcp.getTokenLocation());
+                        created = null;
+                    }
+                    break;
+                case MODIFIED_FIELD:
+                    if (xcp.currentToken() == XContentParser.Token.VALUE_NULL) {
+                        modified = null;
+                    } else if (xcp.currentToken().isValue()) {
+                        modified = Instant.ofEpochMilli(xcp.longValue());
+                    } else {
+                        XContentParserUtils.throwUnknownToken(xcp.currentToken(), xcp.getTokenLocation());
+                        modified = null;
+                    }
+                    break;
+                case DESCRIPTION_FIELD:
+                    description = xcp.text();
+                    break;
+                case LABELS_FIELD:
+                    XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_ARRAY, xcp.currentToken(), xcp);
+                    while (xcp.nextToken() != XContentParser.Token.END_ARRAY) {
+                        String entry = xcp.textOrNull();
+                        if (entry != null) {
+                            labels.add(entry);
+                        }
+                    }
+                    break;
+                case FEED_ID_FIELD:
+                    feedId = xcp.text();
+                    break;
+                default:
+                    xcp.skipChildren();
+            }
+        }
+
+        return new IocDao(
+                id,
+                name,
+                type,
+                value,
+                severity,
+                specVersion,
+                created,
+                modified,
+                description,
+                labels,
+                feedId
+        );
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public IocType getType() {
+        return type;
+    }
+
+    public List<String> getValue() {
+        return value;
+    }
+
+    public String getSeverity() {
+        return severity;
+    }
+
+    public String getSpecVersion() {
+        return specVersion;
+    }
+
+    public Instant getCreated() {
+        return created;
+    }
+
+    public Instant getModified() {
+        return modified;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public List<String> getLabels() {
+        return labels;
+    }
+
+    public String getFeedId() {
+        return feedId;
+    }
+
+    public enum IocType {
+        DOMAIN("domain-name"),
```

Comment:
These are the references I'm using.

##### Domain documentation
Section "6.4 Domain Name Object" of https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_k017w16zutw

##### Hash documentation
TBD - Still looking for a good reference for this

##### IP documentation
"6.8 IPv4 Address Object" section of https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_wh296fiwpklp